### PR TITLE
Use more specific PHPDoc

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is pedantic.

## Changelog

```markdown
### Changed
- Match PHPDoc with doctrine model 

### Removed
- Remove superfluous PHPDoc 
```

### CODE REVIEW

File: src/Entity/BlockManager.php [hunk 1]
```diff
@@ -43,8 +40,6 @@ public function save($block, $andFlush = true)
      * @param int  $parentId Parent block Id (needed when partial = true)
      * @param int  $pageId   Page Id (needed when partial = true)
      * @param bool $partial  Should we use partial references? (Better for performance, but can lead to query issues.)
-     *
-     * @return mixed
```
Comment (phansys) [inline src/Entity/BlockManager.php]: If we will remove the `mixed` types, we should update the rule at [dev-kit](https://github.com/sonata-project/dev-kit/blob/165dd7bb80c3e3a6ce69f0f8f440dd737ea25a47/project/.php_cs.dist.twig#L31).

File: src/Model/SnapshotInterface.php [hunk 2]
```diff
@@ -17,6 +17,13 @@
  * Defines methods to interact with the persistency layer of a SnapshotInterface.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @method int|null           getTargetId()
```
Comment (phansys) [inline src/Model/SnapshotInterface.php]: Are these methods intended to be part of the interface in a next major release?

Comment (core23) [reply src/Model/SnapshotInterface.php]: We should add all @methods in the next major version

File: src/Entity/BlockManager.php (hunk 1 above)
Comment (core23) [reply src/Entity/BlockManager.php]: IMHO we could leave the settings as they are. It is no problem to keep some mixin settings. I reverted some of my changes.

File: tests/Block/BlockContextManagerTest.php [hunk 3]
```diff
@@ -28,7 +28,7 @@ class BlockContextManagerTest extends TestCase
 {
     public function testGetWithValidData()
     {
-        $service = $this->getMockForAbstractClass(AbstractBlockService::class);
+        $service = $this->getMockBuilder(AbstractBlockService::class)->disableOriginalConstructor()->getMock();
```
Comment (OskarStark) [inline tests/Block/BlockContextManagerTest.php]: Why not createMock?

Comment (core23) [reply tests/Block/BlockContextManagerTest.php]: Fixed

**Pull request merged**